src/net_processing.cpp

```diff
@@ -4420,7 +4532,10 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                     // protocol for getting all unconfirmed parents.
                     const auto gtxid{GenTxid::Txid(parent_txid)};
                     AddKnownTx(*peer, parent_txid);
-                    if (!AlreadyHaveTx(gtxid, /*include_reconsiderable=*/true)) AddTxAnnouncement(pfrom, gtxid, current_time);
+                    // Skip m_recent_rejects_reconsiderable because the missing parent may have been
+                    // previously rejected for being too low feerate, and this orphan may be able to
+                    // CPFP it if we consider them as a package.
```

Comment:
this change also only matters for the non-segwit case since we're checking via txid, correct?

Reply:
Correct. If it's a witness tx, we would have cached the error by wtxid but the request is by txid, so this would miss it.